### PR TITLE
Fixes #53 - Unified waiting for ports accross providers

### LINKED ISSUE #53: Unify waiting for start ports
Move waiting for start ports from Node constructor to the cloud provider before bootScript handler.

We should also add possibility to wait for ports after a bootScript is executed.

### PR DESCRIPTION
**Pull request by kwart**
Fixes #53 .

It also
* introduces new properties for after the boot script execution waiting;
* removes deprecated azure.userData* properties

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 ## 0.10.0 (not yet released)
 
+- unified `waitForPorts` and `waitForPorts.timeoutSec` properties across providers
+- added possibility to wait for ports before a boot script is executed; it's configured by
+  `bootScript.waitForPorts` and `bootScript.waitForPorts.timeoutSec` properties
+- removed support for deprecated Azure Node properties `azure.userData`
```
Comment (kwart) [inline CHANGELOG.md]: It was deprecated in `0.9.0` and should be removed in `0.10.0`. IMO it should be fine.

File: core/src/main/java/org/wildfly/extras/sunstone/api/impl/NodeConfigData.java [hunk 2]
```diff
@@ -5,13 +5,8 @@
  * data between a cloud-provider-specific subclass and general {@link AbstractJCloudsNode}.
  */
 public final class NodeConfigData {
```
Comment (kwart) [inline core/src/main/java/org/wildfly/extras/sunstone/api/impl/NodeConfigData.java]: I'm thinking about removing the class completely. IMO it's cleaner than to keep it for just-in-case.

Comment (kwart) [inline core/src/main/java/org/wildfly/extras/sunstone/api/impl/NodeConfigData.java]: Added commit, which removes it.

File: CHANGELOG.md (hunk 1 above)
Comment (LittleJohnII) [inline CHANGELOG.md]: Ah I see, I've been reading the changelog wrong. OK then.

File: core/src/main/java/org/wildfly/extras/sunstone/api/impl/NodeConfigData.java (hunk 2 above)
Comment (LittleJohnII) [inline core/src/main/java/org/wildfly/extras/sunstone/api/impl/NodeConfigData.java]: Fine by me.

**Comment by LittleJohnII** [comment]
LGTM, will merge once the checks are complete.

**Comment by kwart** [comment]
Thanks for review. I'm going to squash the fix-up commit.

**Pull request merged**